src/components/ItemDetails.jsx

```diff
@@ -0,0 +1,22 @@
+import { useParams } from "react-router-dom";
+import styles from "./ItemDetails.module.css";
+
+export const ItemDetails = ({ itemsList }) => {
+  const { id } = useParams();
+  const renderItemDetails = function () {
+    if (itemsList.length === 0) return;
+    const choseItem = itemsList.filter((item) => item.id === id)[0];
```

Comment:
typo in word `choseItem`, should be `chosenItem`.

It would be more elegant to use [useLoaderData](https://reactrouter.com/en/main/hooks/use-loader-data) same way they use it in https://reactrouter.com/en/main/start/tutorial
in order to avoid passing all items into ItemDetails. I'd say its more important to have an elegant solution, then to rush with more features. The smaller the application the less it costs to make changes.

Reply:
Or you could refetch your resource again based on ID( this would be quite standard for a wen-app). Eg: fetch("/myresource/1"). json-server already supports that.